alpha/serial_cmd.cpp

```diff
@@ -209,6 +217,72 @@ static int do_clock_cmd(int argc, char **argv)
   return ESP_OK;
 }
 
+static struct
+{
+  struct arg_lit *get;
+  struct arg_str *ssid;
+  struct arg_str *password;
+  struct arg_end *end;
+} wifi_args;
+
+static int do_wifi_cmd(int argc, char **argv)
+{
+  int nerrors = arg_parse(argc, argv, (void**) &wifi_args);
+
+  if (nerrors != 0)
+  {
+    arg_print_errors(stderr, wifi_args.end, argv[0]);
+    return ESP_ERR_INVALID_ARG;
+  }
+
+  /* get settings from eeprom */
+  TS_Settings* settings = TS_Storage.settings_get();
+
+  /* user issued get cmd */
+  if (wifi_args.get->count == 1)
+  {
+    printf("SSID: %s\n", settings->wifiSsid);
+    printf("Password: %s\n\n", settings->wifiPass);
+  }
+  /* user issued set cmd, store string if length check passes */
+  else if (wifi_args.ssid->count == 1 || wifi_args.password->count == 1)
+  {
+    if (wifi_args.ssid->count == 1)
+    {
+      size_t ssid_strlen = strlen(wifi_args.ssid->sval[0]);
+      if (ssid_strlen <= STR_ARG_MAXLEN)
+      {
+        strcpy(settings->wifiSsid, wifi_args.ssid->sval[0]);
+      }
+      else
+      {
+        printf("SSID exceeds max string length of %d\n\n", STR_ARG_MAXLEN);
+      }
+    }
+    if (wifi_args.password->count == 1)
+    {
+      size_t pass_strlen = strlen(wifi_args.password->sval[0]);
+      if (pass_strlen <= STR_ARG_MAXLEN)
+      {
+        strcpy(settings->wifiPass, wifi_args.password->sval[0]);
+      }
+      else
+      {
+        printf("Password exceeds max string length of %d\n\n", STR_ARG_MAXLEN);
+      }
+    }
+    TS_Storage.settings_save();
```

Comment:
Oh wait, can you make save a separate command?

We need to be able to change settings w/o saving to flash. This is particularly useful during automated testing.

Reply:
Got it. So perhaps default cmd saves to EEPROM and --ram saves to RAM

Reply:
ah no, just have an explicit command that saves

we actually have a settings_load to overwrite whatever is in RAM w/ whats in flash

Reply:
your method works